common/libs/transaction.js

```diff
@@ -29,3 +39,95 @@ export type BroadcastTransaction = {|
   rawTx: string,
   signedTx: string
 |};
+
+export async function generateTransaction(
+  node: BaseNode,
+  tx: Transaction,
+  wallet: BaseWallet,
+  token: ?Token
```

Comment:
Yep, as I've been working my way down, I'm realizing this wants to go damn near everywhere. I'll see if it makes sense to sock it in `Transaction` as an optional key.